oopsypad/server/models.py

```diff
@@ -377,30 +391,35 @@ class Issue(mongo.Document):
 
     @property
     def last_seen(self):
-        minidumps = Minidump.objects(
-            product=self.product,
-            version=self.version,
-            platform=self.platform,
-            crash_reason=self.reason,
-            crash_location=self.location
-        ).only('date_created').order_by('-date_created').first()
-        return minidumps.date_created
+        minidump = self.get_minidumps().only('date_created').order_by(
+            '-date_created').first()
+        if minidump:
+            return minidump.date_created
 
     @property
     def avg_uptime(self):
+        minidumps = self.get_minidumps().filter(process_uptime__exists=True)
```

Comment:
Please use https://docs.mongodb.com/manual/reference/operator/aggregation/avg/ to calculate avg